### PR TITLE
A tool for Party A to request digital signed permission to ask for data about user from Party B

### PR DESCRIPTION
**Pull request by rafaelbernard**
https://app.asana.com/0/home/1178974171608696/1200293282989590

- Get email and name
- Sign the consent message with the IDNZ card number
- Confirm the consent message to disclose the name and email validating the IDNZ card used to consent
- Save consents on local storage

image

image

image

IDCard number not in the wallet
image

IDCard number in the wallet
image

### CODE REVIEW

File: src/id-card/sign-permission-confirmation.html [hunk 1]
```diff
@@ -0,0 +1,196 @@
+<!doctype html>
+<html lang="en-NZ">
+<head>
+    <title>InternetNZ | IRMA examples - ID-Card</title>
```
Comment (ehsan-fj) [inline src/id-card/sign-permission-confirmation.html]: Maybe a more meaningful title? something includes `signing confirmation`?

File: src/id-card/sign-permission.html [hunk 2]
```diff
@@ -0,0 +1,272 @@
+<!doctype html>
+<html lang="en-NZ">
+<head>
+    <title>InternetNZ | IRMA examples - ID-Card</title>
```
Comment (ehsan-fj) [inline src/id-card/sign-permission.html]: again, a more related title would be great.

**Review by ehsan-fj — APPROVED** [review]

File: src/id-card/sign-permission-confirmation.html [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        // --------------------------------
+        // globals loaded from globals file
+        // --------------------------------
+    </script>
+    <script type="text/javascript">
+        window.onload = function () {
+            const IdCardApp = {
+                data() {
+                    return {
+                        CONSENT_MESSAGE: 'I consent IDNZ company to share my name and email with PartyB company.',
```
Comment (camfindlay) [inline src/id-card/sign-permission-confirmation.html]: update this to read "I consent IDNZ to share my disclosed name and email with Centrality for the purpose of associating my IDNZ card with a CENNZnet account". I'm just thinking about how we make this feel a bit more useful... I can imagine a use case where your ID could be associated with your blockchain address/account for example.

Comment (camfindlay) [reply src/id-card/sign-permission-confirmation.html]: if you can change any other references to Party B in the app too please.

**Review by camfindlay — CHANGES_REQUESTED** [review]
change party b to make it feel a bit more real.

File: src/id-card/sign-permission-confirmation.html (hunk 3 above)
Comment (rafaelbernard) [reply src/id-card/sign-permission-confirmation.html]: So we then naming PartyB as Centrality, right?

I thought Centrality was going to be only our storage method, but that is fine if we can reference them as the Party involved in using the info.

Comment (camfindlay) [reply src/id-card/sign-permission-confirmation.html]: This is not a technical thing, it's a scenario to help people understand the point of what we are doing :)

**Pull request merged**